ls1intum/Artemis
# Leaderboards - StudentScore

### Checklist

### Motivation and Context
To make the leaderboards for my bachelor thesis more efficient we want to have the scores of the students closer with the exercises themselves.
Also see Issue #2044.

### Description
There will be two new tables added to the database. The student_score table will include the most recent result and score of an exercise for a student.

### Steps for Testing
best tested locally because currently there are no changes that can be seen or are sent to the client
1. Create exercises and participate in them with different users.
2. Assess the submissions.
3. Have a look in your db and check if all StudentScores are entered correctly.
4. Change assessments.
5. Step 3.

### Test Coverage
- StudentScore: 100%
- StudentScoreResource: 100%
- StudentScoreService: 93%
